github3/session.py

```diff
@@ -37,6 +37,20 @@ def __init__(self):
         self.two_factor_auth_cb = None
         self.request_counter = 0
 
+    def _username(self):
```

Comment:
@sigmavirus24 how do you feel about this addition? It's not the cleanest, though it also isn't too much added complexity.

I contacted GitHub to ask why `username` is required (maybe they'll make a change) after verifying that only the authenticated username can be provided there.